$NetBSD: patch-Makefiles_Makefile.SDL.UNIX.GCC,v 1.2 2012/12/20 14:30:46 othyro Exp $

Add missing dependencies, and fix CFLAGS and LDFLAGS for GL and GLU.

--- Makefiles/Makefile.SDL.UNIX.GCC.orig	2012-01-11 16:47:40.000000000 +0000
+++ Makefiles/Makefile.SDL.UNIX.GCC
@@ -45,6 +45,12 @@ ENABLE_DEBUGGER = no
 ###############################################################################
 
 #
+# GL, GLU
+#
+GLU_CFLAGS = `pkg-config gl glu --cflags`
+GLU_LIBFLAGS = `pkg-config gl glu --libs`
+
+#
 # SDL
 #
 SDL_CFLAGS = `sdl-config --cflags`
@@ -65,12 +71,10 @@ OUTFILE = $(BIN_DIR)/Supermodel
 # Compiler and Linker
 ###############################################################################
 
-CC = gcc
-LD = gcc
-COMPILER_FLAGS = $(SDL_CFLAGS) -ISrc/ -ISrc/OSD/SDL/ -ISrc/Pkgs/ -c -Wall -O3
-CFLAGS = $(COMPILER_FLAGS)
-CPPFLAGS = $(COMPILER_FLAGS)
-LFLAGS = -o $(OUTFILE) $(SDL_LIBFLAGS) -lGL -lGLU -lz -lm -lstdc++
+COMPILER_FLAGS = $(GLU_CFLAGS) $(SDL_CFLAGS) -ISrc/ -ISrc/OSD/SDL/ -ISrc/Pkgs/ -c -Wall
+CFLAGS += $(COMPILER_FLAGS)
+CPPFLAGS = $(CXXFLAGS) $(COMPILER_FLAGS) -std=c++98
+LFLAGS = $(LDFLAGS) -o $(OUTFILE) $(GLU_LIBFLAGS) $(SDL_LIBFLAGS) -lGL -lGLU -lz -lm -lstdc++
 
 #
 # Build options...
@@ -116,7 +120,7 @@ endif
 # Targets
 #
 all:	$(BIN_DIR) $(OBJ_DIR) $(OBJ)
-	$(LD) $(OBJ_LIBS) $(OBJ) $(LFLAGS)
+	$(CC) $(OBJ_LIBS) $(OBJ) $(LFLAGS)
 
 $(BIN_DIR):
 	mkdir $(BIN_DIR) 
@@ -136,9 +140,9 @@ clean:	
 #
 $(OBJ_DIR)/m68kmake: Src/CPU/68K/Musashi/m68kmake.c
 	$(CC) $< $(CFLAGS) -o $(OBJ_DIR)/m68kmake.o
-	$(LD) $(OBJ_DIR)/m68kmake.o -o $(OBJ_DIR)/m68kmake
+	$(CC) $(OBJ_DIR)/m68kmake.o -o $(OBJ_DIR)/m68kmake
 	
-$(OBJ_DIR)/m68kops.h: $(OBJ_DIR)/m68kmake Src/CPU/68K/Musashi/m68k_in.c
+$(OBJ_DIR)/m68kops.h $(OBJ_DIR)/m68kopnz.c $(OBJ_DIR)/m68kopdm.c $(OBJ_DIR)/m68kopac.c $(OBJ_DIR)/m68kops.c: $(OBJ_DIR)/m68kmake Src/CPU/68K/Musashi/m68k_in.c
 	$(OBJ_DIR)/m68kmake $(OBJ_DIR) Src/CPU/68K/Musashi/m68k_in.c
 
 $(OBJ_DIR)/m68kcpu.o: Src/CPU/68K/Musashi/m68kcpu.c $(OBJ_DIR)/m68kops.h Src/CPU/68K/Musashi/m68k.h Src/CPU/68K/Musashi/m68kconf.h Src/CPU/68K/Musashi/m68kctx.h
